api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -410,7 +410,7 @@ private void checkAndAddPartitionName(String name, Integer sourceColumnId) {
         }
       }
       Preconditions.checkArgument(
-          name != null && !name.isEmpty(), "Cannot use empty or null partition name: %s", name);
+          !name.isEmpty(), "Cannot use empty or null partition name: %s", name);
```

Comment:
we can remove "or null" text from message also.

Reply:
+1 on updating the error msg here

Reply:
this still needs to be addressed